app/code/Transoft/Blog/Block/Bloglist.php

```diff
@@ -0,0 +1,137 @@
+<?php
+declare(strict_types=1);
+
+namespace Transoft\Blog\Block;
+
+use Magento\Catalog\Model\Locator\RegistryLocator;
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Framework\Api\SortOrderBuilder;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\Framework\App\RequestInterface;
+use Magento\Framework\Exception\NotFoundException;
+use Magento\Framework\UrlInterface;
+use Magento\Framework\View\Element\Block\ArgumentInterface;
+use Magento\Framework\View\Element\Template;
+use Transoft\Blog\Model\ModelRepository;
+
+/**
+ * Generates blog list
+ */
+class Bloglist extends Template implements ArgumentInterface
```

Comment:
You should call parent ( Magento\Framework\View\Element\Template ) constructor and define parent constructor arguments. ArgumentInterface is used to define View Model and should not be used in Block

Reply:
fixed